src/browser/provider/built-in/base.js

```diff
@@ -0,0 +1,75 @@
+import browserNatives from 'testcafe-browser-natives';
+import OS from 'os-family';
+import WARNING_MESSAGE from '../../../warnings/message';
+
+
+/*eslint-disable no-undef*/
+function getDataFromBrowser () {
+    return { width: window.innerWidth, height: window.innerHeight, title: document.title };
+}
+/*eslint-disable no-undef*/
+
+const initScript = getDataFromBrowser.toString();
+
+
+export default class BaseBrowserProvider {
+    constructor () {
+        // HACK: Browser window has different border size in normal and maximized modes. So we need to be sure that window is
+        // not maximized before resizing in order to keep client area size correction mechanism working. When browser is started,
+        // we are resizing it first time to switch the window to normal mode, and second - to restore the client area size.
+        this.resizeCorrections = {};
+        this.useResizeHack     = OS.win;
+    }
+
+    async _calculateResizeCorrections (id) {
+        await this.waitForConnectionReady(id);
+
+        var { width, height, title } = await this.runInitScript(id, initScript);
+
+        if (!await browserNatives.isMaximized(title))
+            return;
+
+        await browserNatives.resize(title, width, height, width, height);
+
+        var { width: newWidth, height: newHeight } = await this.runInitScript(id, initScript);
+
+        await browserNatives.maximize(title);
+
+        if (newWidth === width && newHeight === height)
+            return;
+
+        this.resizeCorrections[id] = { width: newWidth - width, height: newHeight - height };
+    }
+
+    async openBrowser (id) {
+        if (this.useResizeHack)
```

Comment:
No need for additional var here, it's quite clear that if we have `OS.win` we should use resize correction